code/core/src/telemetry/exec-command-count-lines.ts

```diff
@@ -0,0 +1,33 @@
+import { createInterface } from 'node:readline';
+
+// eslint-disable-next-line depend/ban-dependencies
+import { execaCommand } from 'execa';
+
+/**
+ * Execute a command in the local terminal and count the lines in the result
+ *
+ * @param command The command to execute.
+ * @param options Execa options
+ * @returns The number of lines the command returned
+ */
+export async function execCommandCountLines(
+  command: string,
+  options?: Parameters<typeof execaCommand>[1]
+) {
+  const process = execaCommand(command, { shell: true, buffer: false, ...options });
+  if (!process.stdout) {
+    // Return undefined rather than throwing an error
+    return undefined;
+  }
+
+  let lineCount = 0;
+  const rl = createInterface(process.stdout);
+  rl.on('line', () => {
+    lineCount += 1;
+  });
+
+  // If the process errors, this will throw
+  await process;
```

Comment:
i assume we're guaranteed to get all the lines before this exits?

Reply:
Yes, the "promise" doesn't resolve until the command is done. I guess theorectically the final `on('line')` might fire after that but I've never observed it in practice. I say theorectically just because I don't know for sure either way.